packages/core/src/execution-engine/__tests__/workflow-execute.test.ts

```diff
@@ -246,6 +246,10 @@ describe('WorkflowExecute', () => {
 	});
 
 	describe('runPartialWorkflow2', () => {
+		beforeEach(() => {
+			jest.clearAllMocks();
```

Comment:
considering that we often use `spyOn`, use a custom implementation, or sometimes directly assign method values, I think we should switch to `resetAllMocks` instead where ever possible to ensure no mocks are implicitly shared between tests.

Reply:
Had to amend some other tests and do the mocking in beforeEach blocks.